### PR TITLE
Add CSGO SRCDS input plugin

### PR DESCRIPTION
**Pull request by oofdog**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by zak-pawel** [comment]
@oofdog Let me know when plugin is finished (don't forget to add proper `README.md` and some basic unit tests). I'd love to use it with my server :)

**Comment by oofdog** [comment]
@zak-pawel Ready to go.

File: plugins/inputs/csgo/csgo.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+type CSGO struct {
+	Servers [][]string `toml:"servers"`
+}
+
+func (_ *CSGO) Description() string {
+	return "Fetch metrics from a CSGO SRCDS"
+}
+
+var sampleConfig = `
+  ## specify servers using the following format:
```
Comment (zak-pawel) [inline plugins/inputs/csgo/csgo.go:38]: `Specify`

File: plugins/inputs/csgo/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+The `csgo` plugin gather metrics from CSGO servers.
+
+#### Configuration
+```toml
+[[inputs.csgo]]
+  servers = [
+    ["ip1:port1", "rcon_password1"],
+    ["ip2:port2", "rcon_password2"],
+  ]
+```
```
Comment (zak-pawel) [inline plugins/inputs/csgo/README.md:12]: `SampleConfig must match the readme, but not include the plugin name.` -> https://github.com/influxdata/telegraf/wiki/Review

Please, copy here content of `sampleConfig` from `csgo.go`

File: plugins/inputs/csgo/csgo.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	inputs.Add("csgo", func() telegraf.Input {
+		return &CSGO{}
+	})
+}
+
+func (s *CSGO) gatherServer(
+	server []string,
+	request func(string, string) (string, error),
+	acc telegraf.Accumulator) error {
+	if len(server) != 2 {
```
Comment (zak-pawel) [inline plugins/inputs/csgo/csgo.go:78]: New line for better readability?

File: plugins/inputs/csgo/csgo.go [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+		return errors.New("wrong argument length")
```
Comment (zak-pawel) [inline plugins/inputs/csgo/csgo.go:79]: Can you elaborate on that in this error?

File: plugins/inputs/csgo/csgo.go [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+	err = json.Unmarshal(marshalled, &statsMap)
+	if err != nil {
+		return err
+	}
+
+	acc.AddGauge("csgo", statsMap, tags, now)
+	return nil
+}
+
+func request_server(url string, rconPw string) (string, error) {
```
Comment (zak-pawel) [inline plugins/inputs/csgo/csgo.go:167]: Use camelCase.

File: plugins/inputs/csgo/csgo_test.go [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+	assert.Equal(t, expectedOutput.UptimeMinutes, acc.Metrics[0].Fields["uptime_minutes"])
+	assert.Equal(t, expectedOutput.Maps, acc.Metrics[0].Fields["maps"])
+	assert.Equal(t, expectedOutput.FPS, acc.Metrics[0].Fields["fps"])
+	assert.Equal(t, expectedOutput.Players, acc.Metrics[0].Fields["players"])
+	assert.Equal(t, expectedOutput.Sim, acc.Metrics[0].Fields["sv_ms"])
+	assert.Equal(t, expectedOutput.Variance, acc.Metrics[0].Fields["variance_ms"])
+	assert.Equal(t, expectedOutput.Tick, acc.Metrics[0].Fields["tick_ms"])
+}
+
+func request_mock(_ string, _ string) (string, error) {
```
Comment (zak-pawel) [inline plugins/inputs/csgo/csgo_test.go:44]: camelCase.

File: plugins/inputs/csgo/csgo.go [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+		return err
+	}
+
+	rows := strings.Split(resp, "\n")
+	if len(rows) < 2 {
+		return errors.New("bad response")
+	}
+
+	fields := strings.Fields(rows[1])
+	cpu, err := strconv.ParseFloat(fields[0], 32)
```
Comment (zak-pawel) [inline plugins/inputs/csgo/csgo.go:94]: Check `if len(fields) == 10`

**Review by zak-pawel — CHANGES_REQUESTED** [review]

**Review by zak-pawel — APPROVED** [review]

File: plugins/inputs/csgo/csgo.go [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+	})
+}
+
+func (s *CSGO) gatherServer(
+	server []string,
+	request func(string, string) (string, error),
+	acc telegraf.Accumulator) error {
+
+	if len(server) != 2 {
+		return errors.New("incorrect server config")
```
Comment (Jagularr) [inline plugins/inputs/csgo/csgo.go:80]: might want to make this more descriptive of the issue

File: plugins/inputs/csgo/csgo.go [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+	}
+
+	rows := strings.Split(resp, "\n")
+	if len(rows) < 2 {
+		return errors.New("bad response")
+	}
+
+	fields := strings.Fields(rows[1])
+	if len(fields) != 10 {
+		return errors.New("bad response")
```
Comment (Jagularr) [inline plugins/inputs/csgo/csgo.go:96]: same here, more specificity in the messages might be helpful for the user.

File: plugins/inputs/csgo/csgo.go [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/james4k/rcon"
+)
+
+type statsData struct {
```
Comment (Jagularr) [inline plugins/inputs/csgo/csgo.go:16]: nitpick but `statsData` is redundant, `stats` is as good of a name.

File: plugins/inputs/csgo/csgo.go [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+		return err
+	}
+	msVar, err := strconv.ParseFloat(fields[8], 64)
+	if err != nil {
+		return err
+	}
+	tick, err := strconv.ParseFloat(fields[9], 64)
+	if err != nil {
+		return err
+	}
```
Comment (Jagularr) [inline plugins/inputs/csgo/csgo.go:138]: if you want to get rid of this really long block of almost duplicated code, you could stick it in a loop and add each entry to a separate float64 slice, and then use each index of that slice, respectively, when you create the `stats` object below.

**Review by Jagularr — COMMENTED** [review]
Overall looks good, minor comments.

**Review by ssoroka — APPROVED** [review]
I'm going to merge and you can address Jagularr's feedback in a subsequent PR.

**Pull request merged**